core/trino-grammar/src/main/antlr4/io/trino/grammar/sql/SqlBase.g4

```diff
@@ -53,8 +53,15 @@ statement
          (COMMENT string)?
          (AUTHORIZATION principal)?
          (WITH properties)?                                            #createCatalog
+    | CREATE CATALOG (IF NOT EXISTS)? target=identifier
+         LIKE source=identifier
```

Comment:
What's the use case for "LIKE ..."? What are its semantics when `WITH` is also provided?

Reply:
The use case is when you need to create `target` catalog similar to a `source` one, overriding some properties. 
E.g. you have postgresql catalog configured, which points to `db1`, and for another database `db2` you want to specify similar, but different catalog. So you have to override only URL/password via `WITH`.

Reply:
LIKE is typically used for pattern matching and not for copying/templating .. so the grammar probably need to be changed. As this stands with LIKE it is counterintuitive to any user.

Reply:
It's already used in CREATE TABLE, this mimics it

Reply:
oh yeah .. you are right. I guess with that in mind it might actually make sense.